common/src/main/scala/net/psforever/objects/serverobject/structures/BuildingControl.scala

```diff
@@ -35,43 +37,62 @@ class BuildingControl(building : Building) extends Actor with FactionAffinityBeh
         building.Amenities.foreach(_.Actor forward FactionAffinity.ConfirmFactionAffinity())
       }
       sender ! FactionAffinity.AssertFactionAffinity(building, faction)
+
+    case Building.AmenityStateChange(obj : SpawnTube) =>
+      if(building.Amenities.contains(obj)) {
+        SendMapUpdate(allClients = true)
+      }
+
+    case Building.AmenityStateChange(obj : Generator) =>
```

Comment:
Can this be merged with the above case by passing in a more generic object class since the two are identical otherwise?

Reply:
For now, they're the same.  You could technically make them part of the same case with `obj : Amenity`; but, I'd still then want to check what kind of `Amenity` they are so we could use these cases for triggered facility updates.